api/contrib/envoy/extensions/filters/network/generic_proxy/codecs/http1/v3/http1.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.generic_proxy.codecs.http1.v3;
+
+import "google/protobuf/wrappers.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.generic_proxy.codecs.http1.v3";
+option java_outer_classname = "Http1Proto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/contrib/envoy/extensions/filters/network/generic_proxy/codecs/http1/v3;http1v3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: HTTP1 codec configuration for Generic Proxy]
+// [#extension: envoy.generic_proxy.codecs.http1]
+
+// Configuration for HTTP codec. This HTTP1 codec is used to parse and serialize HTTP1 messages
+// for the generic proxy filter.
+//
+// .. note::
+//   This codec only supports HTTP1.1 messages and does not support HTTP1.0 messages. And it limits
+//   part of the HTTP1.1 features, such as upgrade, connect, etc.
+//   This codec is mainly designed for the features evaluation of the generic proxy filter. Please
+//   be cautious when using it in production.
+message Http1CodecConfig {
+  // If true, the codec will parse and serialize HTTP1 messages in a single frame per message.
```

Comment:
there is no frame in http1, right? I guess you mean the whole message of http1 request should be received by one shot?

Reply:
The `frame` here means the unique unit that the generic proxy will handle, not the frame of HTTP2.